### PR TITLE
Add checks for empty IP addresses in RTPS

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

**Review by simpsont-oci — APPROVED** [review]

File: tests/transport/best_effort_reader/subscriber.cpp [hunk 1]
```diff
@@ -18,19 +19,36 @@ The test contains 3 writers and 3 readers, associated as follows:
...
+    ser_loc << ACE_OutputCDR::from_boolean(false); // requires inline QoS
+
     publication.remote_reliable_ = true;
     publication.remote_data_.length(1);
     publication.remote_data_[0].transport_type = "rtps_udp";
-    publication.remote_data_[0].data.length(5);
-    for (CORBA::ULong i = 0; i < 5; ++i) {
-      publication.remote_data_[0].data[i] = 0;
-    }
+    message_block_to_sequence (mb_locator, publication.remote_data_[0].data);
```
Comment (mitza-oci) [inline tests/transport/best_effort_reader/subscriber.cpp:51]:
```suggestion
    message_block_to_sequence(mb_locator, publication.remote_data_[0].data);
```

File: tests/transport/rtps/subscriber.cpp [hunk 2]
```diff
@@ -264,15 +265,30 @@ ACE_TMAIN(int argc, ACE_TCHAR* argv[])
...
     AssociationData publication;
     publication.remote_id_ = remote;
     publication.remote_reliable_ = true;
     publication.remote_data_.length(1);
     publication.remote_data_[0].transport_type = "rtps_udp";
-    publication.remote_data_[0].data.length(5);
-    for (CORBA::ULong i = 0; i < 5; ++i) {
-      publication.remote_data_[0].data[i] = 0;
-    }
+    message_block_to_sequence (mb_locator, publication.remote_data_[0].data);
```
Comment (mitza-oci) [inline tests/transport/rtps/subscriber.cpp:291]:
```suggestion
    message_block_to_sequence(mb_locator, publication.remote_data_[0].data);
```

File: tests/transport/rtps_directed_write/subscriber.cpp [hunk 3]
```diff
@@ -147,13 +149,30 @@ void SimpleDataReader::data_received(const ReceivedDataSample& sample)
...
+    ser_loc << ACE_OutputCDR::from_boolean(false); // requires inline QoS
+
     publication.remote_reliable_ = true;
     publication.remote_data_.length(1);
     publication.remote_data_[0].transport_type = "rtps_udp";
     publication.remote_data_[0].data.length(5);
-    for (CORBA::ULong i = 0; i < 5; ++i) {
-      publication.remote_data_[0].data[i] = 0;
-    }
+    message_block_to_sequence (mb_locator, publication.remote_data_[0].data);
```
Comment (mitza-oci) [inline tests/transport/rtps_directed_write/subscriber.cpp:175]:
```suggestion
    message_block_to_sequence(mb_locator, publication.remote_data_[0].data);
```

**Review by mitza-oci — COMMENTED** [review]
This is just in tests so I didn't mark it as "Request changes"

**Pull request merged**